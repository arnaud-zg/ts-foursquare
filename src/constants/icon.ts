export enum EIconSize {
  S = '32',
  M = '44',
  L = '64',
  XL = '88',
}
