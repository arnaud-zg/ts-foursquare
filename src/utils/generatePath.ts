import pathToRegexp from 'path-to-regexp'

const compilePath = (path: string) => pathToRegexp.compile(path)

/**
 * Public API for generating a URL pathname from a path and parameters.
 * https://github.com/ReactTraining/react-router/blob/master/packages/react-router/modules/generatePath.js#L20
 */
export const generatePath = (path = '/', params = {}) =>
  path === '/' ? path : compilePath(path)(params)
