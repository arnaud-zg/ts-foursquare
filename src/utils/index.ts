export * from './icon'
export * from './url'
