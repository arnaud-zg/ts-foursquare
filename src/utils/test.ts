import { NHttpStatuses } from 'http-response-status'

export const mockingFetch = ({ response = {} } = {}) => {
  const mockJsonPromise = Promise.resolve(response)
  const mockFetchPromise = Promise.resolve({
    json: () => mockJsonPromise,
    ok: true,
  })

  // @ts-ignore
  global.fetch = jest.fn().mockImplementation(() => mockFetchPromise)
}

export const mockingFailingNotValidFetch = () => {
  const mockFetchPromise = Promise.resolve({
    ok: true,
    json: null,
  })

  // @ts-ignore
  global.fetch = jest.fn().mockImplementation(() => mockFetchPromise)
}

export const mockingFailingNotFoundFetch = () => {
  const mockFetchPromise = Promise.resolve({
    ok: false,
    status: NHttpStatuses.EClientError.NOT_FOUND,
  })

  // @ts-ignore
  global.fetch = jest.fn().mockImplementation(() => mockFetchPromise)
}
