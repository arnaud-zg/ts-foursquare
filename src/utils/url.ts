interface ILocationSearchParam {
  [key: string]: string | number | boolean | undefined
}

export const getLocationSearch = (
  { param = {} }: { param: ILocationSearchParam } = { param: {} }
) => {
  const locationSearch = Object.keys(param)
    .filter(key => param[key] !== undefined && param[key] !== '')
    .map(
      key => `${encodeURIComponent(key)}=${encodeURIComponent(`${param[key]}`)}`
    )
    .join('&')

  return (!!locationSearch && `?${locationSearch}`) || ''
}

export const getLocationHref = ({
  origin,
  pathname,
  param = {},
}: {
  origin: string
  pathname: string
  param?: ILocationSearchParam
}) => `${origin}${pathname}${getLocationSearch({ param })}`
