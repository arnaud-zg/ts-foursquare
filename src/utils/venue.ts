import { NVenueListed } from '../../types'

export const getVenuesListedGroupKey = (group: NVenueListed.IGroup) =>
  `${group.type}-${group.name}`
