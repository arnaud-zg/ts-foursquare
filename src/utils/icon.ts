import { NIcon } from '../../types/icon.d'
import { EIconSize } from '../constants/icon'

interface IIconUrls {
  [iconSize: string]: string
}

export const getIconUrlList = ({ prefix, suffix }: NIcon.IIcon): IIconUrls =>
  Object.keys(EIconSize).reduce(
    (acc, iconKey) => ({
      ...acc,
      [iconKey]: `${prefix}${
        // Ugly typescript hack
        (EIconSize as { [key: string]: any })[iconKey]
      }${suffix}`,
    }),
    {}
  )
