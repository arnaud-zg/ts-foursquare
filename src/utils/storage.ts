import store from 'store'
import { PROJECT_NAME } from '../constants'
import { CLIENT_ID_KEY, CLIENT_SECRET_KEY } from '../constants/api'

export const getCredentials = () => {
  const data = store.get(PROJECT_NAME) || {}

  const { [CLIENT_ID_KEY]: clientId, [CLIENT_SECRET_KEY]: clientSecret } = data

  return {
    clientId,
    clientSecret,
  }
}

export const setCredentials = ({
  clientId,
  clientSecret,
}: {
  clientId: string
  clientSecret: string
}) => {
  if (clientId && clientSecret) {
    store.set(PROJECT_NAME, {
      [CLIENT_ID_KEY]: clientId,
      [CLIENT_SECRET_KEY]: clientSecret,
    })
  }
}

export const removeCredentials = () => {
  store.remove(PROJECT_NAME)
}
