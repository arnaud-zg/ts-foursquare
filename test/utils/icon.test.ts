import { EIconSize, getIconUrlList } from '../../src/utils/icon'

describe('utils/icon/getIconUrlList', () => {
  it('should get a list of icon url', () => {
    const icon = {
      prefix: 'https://ss3.4sqi.net/img/categories_v2/food/steakhouse_',
      suffix: '.png',
    }

    expect(getIconUrlList(icon)).toMatchSnapshot()
  })
})

describe('utils/icon/EIconSize', () => {
  it('should make a snapshot of EIconSize', () => {
    expect(EIconSize).toMatchSnapshot()
  })
})
