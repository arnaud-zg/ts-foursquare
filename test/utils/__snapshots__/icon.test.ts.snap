// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/icon/EIconSize should make a snapshot of EIconSize 1`] = `
Object {
  "L": "64",
  "M": "44",
  "S": "32",
  "XL": "88",
}
`;

exports[`utils/icon/getIconUrlList should get a list of icon url 1`] = `
Object {
  "L": "https://ss3.4sqi.net/img/categories_v2/food/steakhouse_64.png",
  "M": "https://ss3.4sqi.net/img/categories_v2/food/steakhouse_44.png",
  "S": "https://ss3.4sqi.net/img/categories_v2/food/steakhouse_32.png",
  "XL": "https://ss3.4sqi.net/img/categories_v2/food/steakhouse_88.png",
}
`;
